Bug 1786246 clarify Samples Operator documentation for Jenkins

bmcelvee opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1786246

gabemontero commented:
to fully cover the docs ask in https://bugzilla.redhat.com/show_bug.cgi?id=1786246 @bmcelvee that correlates the following facts:
- the samples imagestreams by default point to images in `registry.redhat.io`
- the jenkins imagestreams, while there are versions of those images in `registry.redhat.io`, point to images in the install payload (which is already in the docs, and which you have further highlighted
- with the current capabilities of registry override field in the samples operator config, setting it to `registry.redhat.io` is
1) redundant / unneeded since they are already pointing to `registry.redhat.io` for everything but jenkins
2) breaks our use of the install payload for the jenkins imagestreams ... if you want the jenkins imagestreams to point to `registry.redhat.io`, you'll need to use `oc tag --source=docker ...` to change the specs for their tags

bmcelvee commented:
Thank you, @gabemontero - this helped a ton! 

> to fully cover the docs ask in https://bugzilla.redhat.com/show_bug.cgi?id=1786246 @bmcelvee that correlates the following facts:
> 

>   * the samples imagestreams by default point to images in `registry.redhat.io` 

Done.
> 
>     * the jenkins imagestreams, while there are versions of those images in `registry.redhat.io`, point to images in the install payload (which is already in the docs, and which you have further highlighted
> 
>     * with the current capabilities of registry override field in the samples operator config, setting it to `registry.redhat.io` is
> 
> 
>     1. redundant / unneeded since they are already pointing to `registry.redhat.io` for everything but jenkins 

Added: Setting the `registryoverride` field in the Sample Operator configuration file to link:registry.redhat.io[registry.redhat.io] is redundant because it is already directed to link:registry.redhat.io[registry.redhat.io] for eveything but Jenkins images and imagestreams. 

(I'm verifying the formatting for registry override.)

> 
>     2. breaks our use of the install payload for the jenkins imagestreams ... if you want the jenkins imagestreams to point to `registry.redhat.io`, you'll need to use `oc tag --source=docker ...` to change the specs for their tags

Added: It also breaks the installation payload for Jenkins imagestreams. 

Just one more question on this last point: if it breaks the install payload, should we give the option to use oc tag --source=docker ...`? Or would it be best to omit that information?

gabemontero commented:
> Thank you, @gabemontero - this helped a ton!
> 
> > to fully cover the docs ask in https://bugzilla.redhat.com/show_bug.cgi?id=1786246 @bmcelvee that correlates the following facts:
> 
> > * the samples imagestreams by default point to images in `registry.redhat.io`
> 
> Done.
> 
> > ```
> > * the jenkins imagestreams, while there are versions of those images in `registry.redhat.io`, point to images in the install payload (which is already in the docs, and which you have further highlighted
> > 
> > * with the current capabilities of registry override field in the samples operator config, setting it to `registry.redhat.io` is
> > 
> > 
> > 1. redundant / unneeded since they are already pointing to `registry.redhat.io` for everything but jenkins 
> > ```
> 
> Added: Setting the `registryoverride` field in the Sample Operator configuration file to link:registry.redhat.io[registry.redhat.io] is redundant because it is already directed to link:registry.redhat.io[registry.redhat.io] for eveything but Jenkins images and imagestreams.
> 
> (I'm verifying the formatting for registry override.)
> 
> > ```
> > 2. breaks our use of the install payload for the jenkins imagestreams ... if you want the jenkins imagestreams to point to `registry.redhat.io`, you'll need to use `oc tag --source=docker ...` to change the specs for their tags
> > ```
> 
> Added: It also breaks the installation payload for Jenkins imagestreams.
> 
> Just one more question on this last point: if it breaks the install payload, should we give the option to use oc tag --source=docker ...`? Or would it be best to omit that information?

Yeah that is a workaround that would allow setting the override field to registry.redhat.io.

That said, since the Jenkins image is supported through OCP, you effective break that support correlation.  Also, that update from `oc tag --source=docker ..` would get replaced when you upgrade anyway.

So good question, but let's omit that information.  Very much feels like a slippery slope.

Now, if they want to create their own jenkins imagestream in the openshift namespace that refrences registry.redhat.io, like "jenkins-dev", etc. that is more viable.  They could even add installed jenkins to the skip list.

I'm still not sure if we want to "encourage" such paths.

How about along with my note here, if you add a comment to the bug for future refernce / just in case, we'll stop there, and leave this content as is for now.  If folks push on it in the future, hopefully we'll remember / be able to find that work around and go from there.

/lgtm

gabemontero commented on modules/installation-restricted-network-samples.adoc:
```diff
@@ -29,6 +32,9 @@ ifdef::samplesoperatoraltreg[]
 Operator.
 endif::[]
 
+Setting the `registryoverride` field in the Sample Operator configuration file to link:registry.redhat.io[registry.redhat.io] is redundant because it is already directed to link:registry.redhat.io[registry.redhat.io] for eveything but Jenkins images and imagestreams. It also breaks the installation payload for Jenkins imagestreams. 
```
Coincidently @bmcelvee another BZ along these lines appears to have come in today.

I'm now now leaning toward explicitly preventing the override of the registry for the jenkins images to 

`"docker.io", "registry.redhat.io", "registry.access.redhat.com", "quay.io", "registry.svc.ci.openshift.org"`

As such, leave this text as is, but add the new sentence: `As such, the sample operator prevents the use of these registries for the jenkins imagestreams:  "docker.io", "registry.redhat.io", "registry.access.redhat.com", "quay.io"`

We'll leave out our internal CI registry.

bmcelvee commented:
Thanks, @gabemontero!

@xiuwang PTAL?

xiuwang commented on modules/installation-restricted-network-samples.adoc:
```diff
@@ -29,6 +32,11 @@ ifdef::samplesoperatoraltreg[]
 Operator.
 endif::[]
 
+Setting the registry override field in the Sample Operator configuration file to link:registry.redhat.io[registry.redhat.io] is redundant because it is already directed to link:registry.redhat.io[registry.redhat.io] for eveything but Jenkins images and imagestreams. It also breaks the installation payload for Jenkins imagestreams.
```
s/eveything/everything

bmcelvee replied:
Oops, thanks!

bmcelvee commented:
@openshift/team-documentation please peer review

mburke5678 commented on modules/installation-restricted-network-samples.adoc:
```diff
@@ -29,6 +32,11 @@ ifdef::samplesoperatoraltreg[]
 Operator.
 endif::[]
 
+Setting the registry override field in the Sample Operator configuration file to link:registry.redhat.io[registry.redhat.io] is redundant because it is already directed to link:registry.redhat.io[registry.redhat.io] for everything but Jenkins images and imagestreams. It also breaks the installation payload for Jenkins imagestreams.
```
Do we define the `registry override` field? Is that the same as the `samplesRegistry` field in Step 4?

xiuwang replied:
IMO, the `registry override` is meaning `samplesRegistry`.
Only `samplesRegistry` in samples operator configuration does this function.

bmcelvee replied:
Thank you for pointing that out! I'm updating it now.

mburke5678 commented on modules/installation-restricted-network-samples.adoc:
```diff
@@ -29,6 +32,11 @@ ifdef::samplesoperatoraltreg[]
+Setting the registry override field in the Sample Operator configuration file to link:registry.redhat.io[registry.redhat.io] is redundant because it is already directed to link:registry.redhat.io[registry.redhat.io] for everything but Jenkins images and imagestreams. It also breaks the installation payload for Jenkins imagestreams.
+
+The Samples Operator prevents the use of the following registries for the Jenkins imagestreams: link:docker.io[docker.io], link:registry.redhat.io[registry.redhat.io], link:registry.access.redhat.com[registry.access.redhat.com], link:quay.io[quay.io].
+====
```
Would the registries look better in a bullet list? Or would that make the node too long?

bmcelvee replied:
I was tempted to put them in a bullet list originally, so I I'll go ahead and do so. It will make it easier to read. Thanks for the suggestion!

mburke5678 commented:
@bmcelvee A question, a nit, and an otherwise LGTM

bmcelvee commented:
/cherrypick enterprise-4.2

bmcelvee commented:
/cherrypick enterprise-4.3

bmcelvee commented:
/cherrypick enterprise-4.4
